parachains/runtimes/assets/statemine/src/lib.rs

```diff
@@ -69,11 +69,13 @@ use xcm_config::{KsmLocation, XcmConfig};
 pub use sp_runtime::BuildStorage;
 
 // Polkadot imports
-use pallet_xcm::{EnsureXcm, IsMajorityOfBody};
+use pallet_xcm::{EnsureXcm, IsMajorityOfBody, IsVoiceOfBody};
 use polkadot_runtime_common::{BlockHashCount, SlowAdjustingFeeUpdate};
 use xcm::latest::BodyId;
 use xcm_executor::XcmExecutor;
 
+use kusama_runtime_constants::xcm::body::{FELLOWS_INDEX, STAKING_ADMIN_INDEX};
```

Comment:
Maybe a DQ, but I see you've defined a `ROOT_INDEX` [here](https://github.com/paritytech/polkadot/pull/6273/files#diff-29b00d2d3967a817d5c82c4dae0a7e8cb10da213b80495f91ed28d3c8ffbf22aR106), and I'm wondering if the Statemine side needs to handle "both" Roots?

My guess is "no", that the gov2 root track would just convert into normal root when calling `xcm.send`, just want to make sure that this will handle messages sent via gov2 root.

Reply:
yes, here we convert parent location if origin type Superuser into Root
https://github.com/paritytech/cumulus/blob/876b196bf2f42ddc0c847c05d2bc370a5fb98be2/parachains/runtimes/assets/statemine/src/xcm_config.rs#L122

And pass Xcm origin type as it is, without any conversion 
https://github.com/paritytech/cumulus/blob/876b196bf2f42ddc0c847c05d2bc370a5fb98be2/parachains/runtimes/assets/statemine/src/xcm_config.rs#L127

The xcm transact should be constructed in different ways, 
in case Fellows origin, xcm origin type should be Xcm, 
for Root, Superuser should be chosen.

Reply:
We also can make, in addition, Transact(OriginType/Xcm && Location(Parent, Here)) -> Root

Reply:
I'd be worried about multiple paths to acquire root origins, should try to keep it simple.